crypto/pkcs12/p12_key.c

```diff
@@ -35,8 +35,6 @@ int PKCS12_key_gen_asc(const char *pass, int passlen, unsigned char *salt,
     }
     ret = PKCS12_key_gen_uni(unipass, uniplen, salt, saltlen,
                              id, iter, n, out, md_type);
-    if (ret <= 0)
-        return 0;
     OPENSSL_clear_free(unipass, uniplen);
     return ret;
```

Comment:
@levitte, in this case (and below) the return code is not actually changed because `PKCS12_key_gen_uni()` returns only 0 or 1.
Nevertheless I've adapted the return expression to address your concern and avoid further discussion.